netdata-installer.sh

```diff
@@ -621,6 +624,72 @@ bundle_libwebsockets() {
 
 bundle_libwebsockets
 
+# -----------------------------------------------------------------------------
+
+build_jsonc() {
+  pushd "${1}" > /dev/null || exit 1
+  run env CFLAGS= CXXFLAGS= LDFLAGS= cmake -DBUILD_SHARED_LIBS=OFF .
+  run env CFLAGS= CXXFLAGS= LDFLAGS= make
+  popd > /dev/null || exit 1
+}
+
+copy_jsonc() {
+  target_dir="${PWD}/externaldeps/jsonc"
+
+  run mkdir -p "${target_dir}" "${target_dir}/json-c" || return 1
+
+  run cp "${1}/libjson-c.a" "${target_dir}/libjson-c.a" || return 1
+  run cp ${1}/*.h "${target_dir}/json-c" || return 1
+}
+
+bundle_jsonc() {
+  if [ -z "${NETDATA_DISABLE_CLOUD}" ] && [ -z "${NETDATA_BUILD_JSON_C}" ] && pkg-config json-c ; then
```

Comment:
I don't think this does what we want?


A | B | C | A & B & C
-- | -- | -- | --
1 | 1 | 1 | 1
0 | 1 | 1 | 0
1 | 0 | 1 | 0
0 | 0 | 1 | 0
1 | 1 | 0 | 0
0 | 1 | 0 | 0
1 | 0 | 0 | 0
0 | 0 | 0 | 0

We effectively only skip "bundling" of `json-c` __iif__:

- `NETDATA_DISABLE_CLOUD` is undefined or a null string.
- `NETDATA_BUILD_JSON_C` is undefined or a null string.
- `pkg-config json-c` returns success / zero.

So...

- If `json-c` is found on the system and we didn't pass the `--disable-cloud` or `--build-json-c` flags we skip bundling our own json-c. 👍
- If `json-c` is __NOT__ found on the system and we didn't pass the `--disable-cloud` or `--build-json-c` flags we bundle our own json-c. 👍
- If we pass in the `--disable-cloud` flag we bundle our own json-c ❌
- If we pass in the `--build-json-c` flag and `json-c` is found on the user's system we bundle our own json-c 👍

...

As discussed over a Slack call the logic falls apart a bit here for `--disable-cloud` and I _think_ we should pslit this expression out and have an early return for this.

As agreed I'l add a commit to your PR and this comment, but I will not be able to then re-review this :)

Reply:
Pushed the afremented commit.